conformance-test: use kubelet healthz port.

hoskeri opened the pull request:
The readonly port could be disabled.

Since we are only using the /healthz endpoint,
we can use the healthz port.

/kind cleanup

#### What this PR does / why we need it:

Removes an unnecessary usage of the kubelet readonly port, replacing it with healthz port instead. 

```release-note
NONE

```

hoskeri committed:
conformance-test: use kubelet healthz port.

The readonly port could be disabled.

Since we are only using the /healthz endpoint,
we can use the healthz port.

Change-Id: If004f2888ca5847b9e2d8c02d5615bed52d94b24

oomichi commented:
Don't we need to update

https://github.com/kubernetes/kubernetes/blob/820247a3aec7d7d1e65cd5ad34b0f07adb16e907/test/e2e_node/util.go#L74

also?

/cc @oomichi

SergeyKanzhelev commented:
/assign

SergeyKanzhelev commented:
/triage accepted
/priority important-longterm

SergeyKanzhelev commented:
/ok-to-test

hoskeri committed:
e2e_node/{service,util}: use kubelet healthz port.

The readonly port could be disabled.

Since we are only using the /healthz endpoint,
we can use the healthz port for this.

Change-Id: Ie0e05a5ab4ec6f51e4d3c63226aa23c1b3a69956

hoskeri commented:
> Don't we need to update
> 
> https://github.com/kubernetes/kubernetes/blob/820247a3aec7d7d1e65cd5ad34b0f07adb16e907/test/e2e_node/util.go#L74
> 
> also?
> 
> /cc @oomichi

Thanks! added another commit that fixes two other locations.

oomichi commented on test/e2e_node/services/kubelet.go:
```diff
@@ -90,10 +90,11 @@ func RunKubelet() {
 const (
 	// Ports of different e2e services.
 	kubeletReadOnlyPort = "10255"
```
We can remove this `kubeletReadOnlyPort` because it is not used after replacing line 97 with `kubeletHealthzPort`.

oomichi reviewed (commented):
Thanks for updating, just one comment for cleanup.

oomichi commented:
Thank you for updating again.

/lgtm

tallclair commented on test/e2e_node/services/kubelet.go:
```diff
@@ -89,11 +89,11 @@ func RunKubelet() {
 	KubeletRootDirectory = "/var/lib/kubelet"
 	// Health check url of kubelet
-	kubeletHealthCheckURL = "http://127.0.0.1:" + kubeletReadOnlyPort + "/healthz"
+	kubeletHealthCheckURL = "http://127.0.0.1:" + kubeletHealthzPort + "/healthz"
```
Can you dedupe this with the one in util.go? I'm not sure which direction the dependencies go, but I'd just pick one to export.

tallclair commented on test/e2e_node/util.go:
```diff
@@ -71,7 +71,8 @@ const (
 	defaultPodResourcesTimeout = 10 * time.Second
 	defaultPodResourcesMaxSize = 1024 * 1024 * 16 // 16 Mb
 	kubeletReadOnlyPort        = "10255"
```
I think this is unused now too?

SergeyKanzhelev replied:
used in line 263

SergeyKanzhelev commented on test/e2e_node/util.go:
```diff
@@ -259,7 +260,7 @@ func logKubeletLatencyMetrics(metricNames ...string) {
 		metricSet.Insert(kubeletmetrics.KubeletSubsystem + "_" + key)
 	}
-	metric, err := e2emetrics.GrabKubeletMetricsWithoutProxy(framework.TestContext.NodeName+":10255", "/metrics")
+	metric, err := e2emetrics.GrabKubeletMetricsWithoutProxy(framework.TestContext.NodeName+":"+kubeletReadOnlyPort, "/metrics")
```
nit: maybe use `JoinHostPort` function here

oomichi commented:
/test pull-kubernetes-e2e-kind
/lgtm

tallclair reviewed (commented):
/lgtm
/approve

hoskeri commented:
/assign @thockin 

Thanks!

thockin reviewed (commented):
Thanks!

/lgtm
/approve
